src/graphql-config/graphql-config.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Env } from '@fullstacksjs/toolbox';
+import type { ApolloDriverConfig } from '@nestjs/apollo';
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import type { GqlOptionsFactory } from '@nestjs/graphql';
+import { ApolloServerPluginLandingPageLocalDefault } from 'apollo-server-core';
+
+import type { EnvironmentVariables } from '../configuration/EnvironmentVariables';
+import { IBAN, Money } from '../shared/scalars';
+
+const mocks = {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  DateTime: () => {
+    return new Date();
+  },
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  String: () => {
+    return 'string-mock';
+  },
+};
+
+@Injectable()
+export class GraphQLConfigService implements GqlOptionsFactory {
+  constructor(private configService: ConfigService<EnvironmentVariables>) {}
+  createGqlOptions(): ApolloDriverConfig {
+    const isDev = Env.isDev;
+    return {
+      playground: false,
+      debug: true,
+      mocks,
+      autoSchemaFile: true,
+      resolvers: { Money, IBAN },
+      plugins: [ApolloServerPluginLandingPageLocalDefault()],
+      introspection: this.configService.get('INTROSPECTION_ENABLED'),
+      cors: {
+        // TODO: origin should be moved into configModule - not sure regex is valid!
+        // https://www.debuggex.com/r/XTrC-yG2Yeq2_sAm
+        origin: isDev
+          ? [
+              /https?:\/\/[a-z0-9\-_]+-fullstacks\.vercel\.app/,
+              'https://studio.apollographql.com/sandbox',
+            ]
+          : [/https?:\/\/[a-z0-9\-_]+-fullstacks\.vercel\.app/],
```

Comment:
@ASafaeirad 
is it possible to use ``@fullstacksjs/toolbox`` here?

Reply:
Of course

Reply:
It's part of devDep, it's ok?

I do some cleanup again and check it.

It's better as I said before to do something for configuration,
we need a cleaner implementation

Reply:
It shouldn't be. change it to normal deps.

Reply:
You mean i changed it to the previous one?

Reply:
No, I mean move `@fullstacksjs/toolbox` to normal dependencies.

Reply:
Done